packages/account/src/Sections/Security/DeactivateAccount/deactivate-account-reason.jsx

```diff
@@ -211,14 +212,35 @@ class DeactivateAccountReason extends React.Component {
         }
     };
 
+    handleChange = (e, onChange) => {
+        const { remaining_characters, total_accumulated_characters, input_action } = this.state;
+
+        if ((remaining_characters <= 0 || total_accumulated_characters >= character_limit_no) && !input_action) {
+            e.preventDefault();
+        } else {
+            onChange(e);
+        }
+    };
+
     handleInputKeyDown = e => {
-        if (this.state.remaining_characters <= 0 && !allowed_keys.has(e.key)) {
+        const key = e.key;
+        let input_action = key;
+
+        if (this.state.remaining_characters <= 0 && !allowed_keys.has(key)) {
+            input_action = null;
```

Comment:
I think instead of using `input_action`, we can add `e.stopPropagation();` to stop the event propagation which will prevent onchange event from being called

Reply:
Hi @balakrishna-binary the `input_action` here is used to log the last keyboard action of the user. Unfortunately we can't prevent user input on `change` event because on android devices there's a bug where `keycode` is always returning same digit on any keystrokes that's why this is handled in `keydown` event. Also the `input_action` is the medium we use to tell the `change` event that the last action is either `backspace` or other special actions and by that user will be able to make a `delete` action even though they have reached the max characters allowed. This `input_action` is not necessary if we will only focus on browser and ios devices but on android it's a different story.